test/collapsers/javascript.js

```diff
@@ -0,0 +1,38 @@
+'use strict';
+var assert = require('power-assert');
+var Bluebird = require('bluebird');
+var describe = require('mocha').describe;
+var it = require('mocha').it;
+var collapser = require('../../lib/collapsers/javascript');
+
+describe('JavaScript collapser', function () {
+  it('should minify JavaScript', function () {
+    return collapser(new Buffer('alert("foo: " + bar)'))
+      .then(function (encoded) {
+        assert(typeof encoded === 'string');
+      });
+  });
+
+  it('should reject if invalid JavaScript', function () {
```

Comment:
Does downstream hold the original JS? Parsing error might just mean that Uglify doesn't support some newer syntax, it's technically not an error, we can just preserve such content unminified.